runelite-client/src/main/java/net/runelite/client/plugins/prayerindicator/PrayerIndicatorOverlay.java

```diff
@@ -0,0 +1,269 @@
+import net.runelite.api.ItemID;
+import net.runelite.api.Query;
+import net.runelite.api.Skill;
+import net.runelite.api.queries.InventoryItemQuery;
+import net.runelite.api.widgets.Widget;
+import net.runelite.api.widgets.WidgetInfo;
+import net.runelite.client.game.ItemManager;
+import net.runelite.client.plugins.itemstats.potions.PrayerPotion;
+import net.runelite.client.ui.overlay.Overlay;
+import net.runelite.client.ui.overlay.OverlayLayer;
+import net.runelite.client.ui.overlay.OverlayPosition;
+import net.runelite.client.ui.overlay.infobox.Counter;
+import net.runelite.client.ui.overlay.infobox.InfoBoxManager;
+import net.runelite.client.util.QueryRunner;
+
+public class PrayerIndicatorOverlay extends Overlay
+{
+	
+	//Set up some constants that will be used later on
+	private final float THICKNESS = 2f;			// This will set how thick the circle should be
+	
+	private Color orbIndicatorColor = new Color (0, 255, 255, 255);
+	private int alphaOffset = -10;			// How much to subtract/add to the alpha of the indicator orb
+	private int playerPotionID;				// The current ItemID of the potion the player is carrying
+	private BufferedImage prayPotIcon;
+	private Counter indicator;				// The indicator that will go in the info box if the player chooses to have it shown
+	
+	//private PrayerIndicatorPlugin plugin;
+	private final Client client;
+	private final PrayerIndicatorConfig config;
+	private final InfoBoxManager infoBoxManager;
+	private final QueryRunner queryRunner;
+	
+	@Inject
+	private ItemManager itemManager;
+	
+	@Inject
+	private PrayerIndicatorPlugin plugin;
+	
+	@Inject
+	public PrayerIndicatorOverlay (QueryRunner queryRunner, Client client, PrayerIndicatorConfig config, InfoBoxManager infoBoxManager)
+	{
+		setPosition (OverlayPosition.DYNAMIC);
+		setLayer (OverlayLayer.ABOVE_WIDGETS);
+		this.client = client;
+		this.config = config;
+		this.infoBoxManager = infoBoxManager;
+		this.queryRunner = queryRunner;
+	}
+	
+	// This method will be called every game tick and will be used to fade the orb indicator in and out
+	public void onTick ()
+	{
+		//Every game tick let's get what kind of prayer potion the player has
+		playerPotionID = getPlayerPotion ();
+		
+		//Let's make sure the infoBox indicator is set
+		if (indicator == null)
+		{
+			prayPotIcon = itemManager.getImage (ItemID.PRAYER_POTION4);
+			indicator = new Counter (prayPotIcon, plugin, "");
+		}
+	}
+	
+	// This method will go every time the config was changed and is only used for if the player deactivates the info box display so it can be properly be removed
+	public void onConfigChanged ()
+	{
+		if (!config.inInfoBox () && infoBoxManager.getInfoBoxes ().contains (indicator))
+			infoBoxManager.removeInfoBox (indicator);
+	}
+	
+	@Override
+	public Dimension render (Graphics2D graphics)
+	{
+		//Before adding it to the infoBox let's make the sure the player want's it to be there
+		if (config.inInfoBox ())
+		{
+			if (isPlayerPrayerLow ())
```

Comment:
For render(), you can call isPlayerPrayerLow() as the first statement, and return null if not. This will reduce the indenting, avoids also checking the config, and it doesn't need to be called twice if the player has both options enabled.